pulp_2to3_migration/app/pre_migration.py

```diff
@@ -224,13 +225,16 @@
         importers_repos = plan.get_importers_repos()
         distributors_repos = plan.get_distributors_repos()
 
+        # filter by repo type
+        repos_to_premigrate = []
+        for repo_type in plan.get_plugins():
+            repos_to_premigrate += type_to_repo_ids[repo_type]
+
         # in case only certain repositories are specified in the migration plan
         if repos:
-            mongo_repo_q &= mongo_Q(repo_id__in=repos)
+            repos_to_premigrate = set(repos).difference(repos_to_premigrate)
```

Comment:
repos_to_premigrate should be converted to set as well.

Reply:
I'm not sure I fully followed. However I think there is no pulp3 repos involved here so there should be no concern about that.

From `type_to_repo_ids[repo_type]` we get pulp 2 repo_ids and from the plan (`repos`) we get pulp 2 repo_ids.
Do you agree?

Reply:
my plan:

```
{
    "plan": {
        "plugins": [
            {
                "repositories": [
                    {
                        "name": "file",
                        "pulp2_importer_repository_id": "file",
                        "repository_versions": [
                            {
                                "pulp2_repository_id": "file"
                            }
                        ]
                    }
                ],
                "type": "iso"
            }
        ]
    },
    "pulp_created": "2020-01-13T13:44:10.661456Z",
    "pulp_href": "/pulp/api/v3/migration-plans/c9cdad50-9a24-4f32-a552-e9486fa73b0a/"
}
```
the repos_to_premigrate would be empty

Reply:
and if i have
```
{
    "plan": {
        "plugins": [
            {
                "repositories": [
                    {
                        "name": "new-name",
                        "pulp2_importer_repository_id": "file",
                        "repository_versions": [
                            {
                                "pulp2_repository_id": "file"
                            }
                        ]
                    }
                ],
                "type": "iso"
            }
        ]
    },
    "pulp_created": "2020-01-13T13:44:10.661456Z",
    "pulp_href": "/pulp/api/v3/migration-plans/c9cdad50-9a24-4f32-a552-e9486fa73b0a/"
}

repos_to_premigrate would container `new-name`, we do not have such repo-id in pulp2 to premigrate

Reply:
Thanks for pointing to the intersection! <3